test/s3.js

```diff
@@ -5,7 +5,12 @@ var s3 = require('../src/js/s3');
 var fs = require('fs');
 var debug = require('gulp-debug');
 var through2 = require('through2').obj;
-var AWS = require('../node_modules/gulp-awspublish/node_modules/aws-sdk');
+var awspath = '../node_modules/gulp-awspublish/node_modules/aws-sdk';
+if (fs.existsSync(__dirname+'/../node_modules/aws-sdk/package.json')) {
+  awspath = '../node_modules/aws-sdk';
+}
+var AWS = require(awspath);
```

Comment:
Oh, right. I missed the last line. 

I think we should just depend on it directly as devDependency. I don't see anything different with this and other devDependencies.